src/link.cpp

```diff
@@ -2760,42 +2770,71 @@ static void construct_linker_job_macho(LinkJob *lj) {
         lj->args.append(lib_dir);
     }
 
+    // .o files
     for (size_t i = 0; i < g->link_objects.length; i += 1) {
         lj->args.append((const char *)buf_ptr(g->link_objects.at(i)));
     }
 
-    // libc++ dep
```

Comment:
didn't we have to link libc++ before certain other deps?

Reply:
not that I'm aware of. Traditionally the more  something is depended on moves later in link lines. These days I don't think that matters much. This new position also matches linux more.